File: Week 2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // Prompt the user for some text
    string text = get_string("Text: ");

    // Count the number of letters, words, and sentences in the text
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    float L = letters / (float) words * 100;
    float S = sentences / (float) words * 100;

    // Compute the Coleman-Liau index
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    // Print the grade level
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }

    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

int count_letters(string text)
{
    int length = strlen(text);
    int counter = 0;
    for (int i = 0; i < length; i++)
    {
        if (isalpha(text[i]) != 0)
        {
            counter++;
        }
    }
    return counter;
}

int count_words(string text)
{
    int length = strlen(text);
    int counter = 1;
    for (int i = 0; i < length; i++)
    {
        if (text[i] == ' ')
        {
            counter++;
        }
    }
    return counter;
}

int count_sentences(string text)
{
    int length = strlen(text);
    int counter = 0;
    for (int i = 0; i < length; i++)
    {
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            counter++;
        }
    }
    return counter;
}

File: week6/sentimental-readability/readability.py
def main():
    text = input("Enter: ")

    words = count(text)[0]
    L = count(text)[1] / words * 100
    S = count(text)[2] / words * 100

    index = round(0.0588 * L - 0.296 * S - 15.8)

    if index < 1:
        print("Before Grade 1")
    elif index > 16:
        print("Grade 16+")
    else:
        print("Grade", index)


def count(text):
    words = 1
    letters = 0
    sentences = 0

    for i in text:
        if i == " ":
            words += 1

        elif i.isalpha():
            letters += 1

        elif i in [".", "!", "?"]:
            sentences += 1

    return words, letters, sentences


if __name__ == "__main__":
    main()
